lumigator/python/mzai/backend/backend/services/experiments.py

```diff
@@ -0,0 +1,48 @@
+import json
+import time
+
+import loguru
+import requests
+from fastapi import BackgroundTasks
+from lumigator_schemas.experiments import ExperimentCreate, ExperimentResponse
+from lumigator_schemas.jobs import (
+    JobEvalCreate,
+)
+
+from backend.services.jobs import JobService
+
+
+class ExperimentService:
+    def __init__(self, job_service: JobService):
+        self._job_service = job_service
+
+    def experiment_specific_background_task(self, job_id: str):
+        status = "PENDING"
+        loguru.logger.info(f"Job id: {job_id}")
+        while status != "SUCCEEDED" and status != "FAILED":
+            res = requests.get(f"http://localhost:8000/api/v1/health/jobs/{job_id}")
+            status = json.loads(res.text)["status"]
+            loguru.logger.info(f"Job id: {job_id}, status: {status}")
+            time.sleep(5)
```

Comment:
I'd suggest making this an async fun and using asyncio.sleep, otherwise the rest should be the same 👍

Reply:
(This is independent of using or not BackgroundTasks, since it seems to use the asyncio event loop anyway under the hood for async funs)

Reply:
Thanks for the suggestion!

Requests does not work well with asyncio, which is why I opted for fastapi managing this with threads. In my next commit I will introduce proper code to get a job's status which relies on the ray client. This is not just way cleaner, but it also seems (just tested!) to be working properly within async so I'll be able to switch to async functions.

One of the advantages of using FastAPI built-in background task management is that it automatically deals with both sync and async functions, so I could first test something quickly with requests and then move to ray client + async.

Reply:
Ouch, I see :-/ Let's tackle this later on, then. Thanks a lot for the info!!! :)